realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1344,6 +1248,13 @@ public void run() {
                                 });
                             }
                         }
+
+                        if (!handlerController.threadContainsAsyncQueries() && !asyncTransactionCallbacks.isEmpty()) {
+                            for (Runnable transactionCallback : asyncTransactionCallbacks) {
+                                handler.post(transactionCallback);
```

Comment:
I am thinking about just call those callbacks in the next `HandlerController.realmChanged()` to make the logic simpler. What do you think?